### PR TITLE
Allow to set labels to ServiceMonitors

### PR DESCRIPTION
**Pull request by canhnt**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

This PR allows Cilium Helm chart to add custom labels to ServiceMonitors: cilium-agent, cilium-operator, hubble.

Fixes: #issue-number

Release note:
```release-note
Allow to add custom labels to ServiceMonitors cilium-agent, cilium-operator, hubble in the Cilium Helm chart.
```

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ kind: ServiceMonitor
 metadata:
   name: cilium-agent
   namespace: {{ .Values.prometheus.serviceMonitor.namespace | default .Release.Namespace }}
+  labels:
+    {{- range $key, $value := .Values.prometheus.serviceMonitor.labels }}
```
Comment (sayboras) [inline install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml]: just a style and mainly for consistency, in other places we are doing something like below.

```
        {{- with .Values.preflight.podLabels }}
        {{- toYaml . | nindent 8 }}
        {{- end }}
```

Comment (canhnt) [reply install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml]: I refactored to your suggested syntax.

**Review by sayboras — APPROVED** [review]
LGTM :100:

File: install/kubernetes/cilium/values.yaml [hunk 2]
```diff
@@ -1058,6 +1061,9 @@ prometheus:
     # This requires the prometheus CRDs to be available (see https://github.com/prometheus-operator/prometheus-operator/blob/master/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml)
     #
     enabled: false
+    # -- Labels to add to ServiceMonitor cilium-agent
+    #
```
Comment (kaworu) [inline install/kubernetes/cilium/values.yaml]: nit: I think we can remove the extra empty comment line, bonus points for removing the one at line 1062 as well.

```suggestion
```

File: install/kubernetes/cilium/values.yaml [hunk 3]
```diff
@@ -1369,6 +1375,9 @@ operator:
       # This requires the prometheus CRDs to be available (see https://github.com/prometheus-operator/prometheus-operator/blob/master/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml)
       ##
       enabled: false
+      # -- Labels to add to ServiceMonitor cilium-operator
+      #
```
Comment (kaworu) [inline install/kubernetes/cilium/values.yaml]: ditto

**Review by kaworu — CHANGES_REQUESTED** [review]
LGTM. Please remove or edit the release-note section of the PR's description.

File: install/kubernetes/cilium/values.yaml (hunk 2 above)
Comment (canhnt) [reply install/kubernetes/cilium/values.yaml]: I see quite many extra empty comment lines in the values.yaml. Should they be removed all for consistency?

Comment (canhnt) [reply install/kubernetes/cilium/values.yaml]: I removed some extra comment lines, please check them out. Some others should be intact.

**Review by kaworu — CHANGES_REQUESTED** [review]
Thanks!

Could you re-run `make -C install/kubernetes docs` and commit the changes to `install/kubernetes/cilium/README.md` please?

**Comment by canhnt** [comment]
> Thanks!
> 
> Could you re-run `make -C install/kubernetes docs` and commit the changes to `install/kubernetes/cilium/README.md` please?

Thank you. I didn't know it's auto-generated.

**Comment by canhnt** [comment]
> LGTM. Please remove or edit the release-note section of the PR's description.

Updated release-note section in the PR's description.

**Comment by kaworu** [comment]
Thank you @canhnt! LGTM, could you squash the commits into one and add a commit description to make checkpatch happy please?

**Comment by canhnt** [comment]
> Thank you @canhnt! LGTM, could you squash the commits into one and add a commit description to make checkpatch happy please?

I squashed all commits in the PR to the single one.

**Review by aanm — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Review by kaworu — APPROVED** [review]

**Review by errordeveloper — APPROVED** [review]

**Comment by canhnt** [comment]
Hi maintainers, 

Is it possible to backport this PR to the branch `1.10.x`? 

It's because we're using Helm chart v1.10.4 (and will use latest 1.10.5) and need to set system-level labels for Cilium ServiceMonitors.

Thanks,
Canh N.

**Pull request merged**